keccak256/src/gates/theta.rs

```diff
@@ -0,0 +1,223 @@
+use halo2::circuit::Layouter;
+use halo2::{
+    circuit::Region,
+    plonk::{Advice, Column, ConstraintSystem, Error, Expression, Selector},
+    poly::Rotation,
+};
+use itertools::Itertools;
+use pasta_curves::arithmetic::FieldExt;
+use std::marker::PhantomData;
+
+#[derive(Clone, Debug)]
+pub struct ThetaConfig<F> {
+    q_enable: Selector,
+    state: [Column<Advice>; 25],
+    _marker: PhantomData<F>,
+}
+
+impl<F: FieldExt> ThetaConfig<F> {
+    pub fn configure(
+        q_enable: Selector,
+        meta: &mut ConstraintSystem<F>,
+        state: [Column<Advice>; 25],
+    ) -> ThetaConfig<F> {
+        meta.create_gate("theta", |meta| {
+            let q_enable = meta.query_selector(q_enable);
+            let column_sum: Vec<Expression<F>> = (0..5)
+                .map(|x| {
+                    let state_x0 =
+                        meta.query_advice(state[5 * x], Rotation::cur());
+                    let state_x1 =
+                        meta.query_advice(state[5 * x + 1], Rotation::cur());
+                    let state_x2 =
+                        meta.query_advice(state[5 * x + 2], Rotation::cur());
+                    let state_x3 =
+                        meta.query_advice(state[5 * x + 3], Rotation::cur());
+                    let state_x4 =
+                        meta.query_advice(state[5 * x + 4], Rotation::cur());
+                    state_x0 + state_x1 + state_x2 + state_x3 + state_x4
+                })
+                .collect::<Vec<_>>();
+
+            (0..5)
+                .cartesian_product(0..5)
+                .map(|(x, y)| {
+                    let new_state =
+                        meta.query_advice(state[5 * x + y], Rotation::next());
+                    let old_state =
+                        meta.query_advice(state[5 * x + y], Rotation::cur());
+                    let right = old_state
+                        + column_sum[(x + 4) % 5].clone()
+                        + Expression::Constant(F::from(13))
```

Comment:
When we hardcode numbers in the impl. Would be nice to add a comment or some info on "Where does this cone from".

In this case would be the check of the base 13 IIUC. But would be nice to add it.

Reply:
Nice catch, I will fix it.